hxd/Pixels.hx

```diff
@@ -536,8 +538,33 @@ class Pixels {
 
 	public static function calcDataSize( width : Int, height : Int, format : PixelFormat ) {
 		return switch( format ) {
-		case S3TC(_):
-			(((height + 3) >> 2) << 2) * calcStride(width, format);
+		case S3TC(n):
+			var w = ((width + 3) >> 2) << 2; // Round up width to next multiple of 4
+			var h = ((height + 3) >> 2) << 2; // Round up height to next multiple of 4
+			var blocks = (w >> 2) * (h >> 2); // Total number of blocks

```

Comment:
This could be simplified without  <<2  and blocks = w*h  (same for other formats)